src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -43,11 +49,61 @@ define([
         deploymentsFetchState: States.STATE_LOADING,
         fetchState: States.STATE_LOADING,
         modalClass: null,
+        route: null,
         tasksFetchState: States.STATE_LOADING
       };
     },
 
     componentDidMount: function() {
+      var router = this.props.router;
+
+      function callRoute(routeFunc) {
+        if(this.state.fetchState === States.STATE_LOADING) {
+          setTimeout(callRoute.bind(this, routeFunc), 300);
```

Comment:
This hacky and probably because you are trying to set state while you are rendering